### PR TITLE
Introduce common option `target`

### LINKED ISSUE #2096: Introduce common options
I think we can make `sider.yml` more user-friendly by introducing common option names.

For example, the stylelint runner supports the `glob` option for analyzed files, on the other hand, the ESLint runner supports the `target` option (previously `dir`).

https://github.com/sider/runners/blob/90a9a49e96fdee49cf74895df8784d618205fb16/lib/runners/processor/stylelint.rb#L16

https://github.com/sider/runners/blob/90a9a49e96fdee49cf74895df8784d618205fb16/lib/runners/processor/eslint.rb#L10-L11

I suggest the following options for common:

- `linter.{id}.target`: files or directories to analyze
- `linter.{id}.dependencies`: e.g. `gems` or `jvm_deps`

For example:

```yaml
linter:
  rubocop:
    target: src/**/*.rb
    dependencies:
      - { name: rubocop-rails, version: 1.2.3 }
  checkstyle:
    target: src/
    dependencies:
      - { name: "com.github.sevntu-checkstyle:sevntu-checks", version: 1.37.1 }
```

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This change introduces a new option `linter.<id>.target`. The option is an alias for the existing option (non-breaking).

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

Close #2096

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: lib/runners/processor/jshint.rb [hunk 1]
```diff
@@ -6,22 +6,23 @@ class Processor::Jshint < Processor
       extend Schema::ConfigTypes
 
       # @type self: SchemaClass
-
       let :config, base(
-        dir: string?,
+        target: target,
+        dir: target, # alias for `target`
```
Comment (ybiquitous) [inline lib/runners/processor/jshint.rb:11]: [note] `jshint` supports multiple targets.

File: lib/runners/processor/stylelint.rb [hunk 2]
```diff
@@ -13,7 +13,8 @@ class Processor::Stylelint < Processor
         'ignore-disables': boolean?,
         'report-needless-disables': boolean?,
         quiet: boolean?,
-        glob: string?,
+        target: target,
+        glob: target, # alias for `target`
```
Comment (ybiquitous) [inline lib/runners/processor/stylelint.rb:17]: [note] `stylelint` supports multiple targets.

File: lib/runners/processor/tyscan.rb [hunk 3]
```diff
@@ -9,7 +9,8 @@ class Processor::Tyscan < Processor
       let :config, npm(
         config: string?,
         tsconfig: string?,
-        paths: array?(string),
+        target: target,
+        paths: target, # alias for `target`
```
Comment (ybiquitous) [inline lib/runners/processor/tyscan.rb:13]: [note] `tyscan` should support a single target.

**Comment by yykamei** [comment]
[question] It makes sense. Is it possible to include `target` in the `base` schema?

**Comment by ybiquitous** [comment]
Oh, sorry. You mean `Schema::ConfigTypes#target`?

**Comment by yykamei** [comment]
Yes, but I'm not sure whether that makes sense...

**Comment by ybiquitous** [comment]
I also could do it, but some linters do not support multiple targets.
For example:

https://github.com/sider/runners/blob/30c1336ab0a18c443fb9f5b9846617368ec7a296/lib/runners/processor/pmd_java.rb#L93

**Comment by yykamei** [comment]
I see. So, overriding in such linters does not work, does it?

**Comment by ybiquitous** [comment]
OK, I'll try it. 💪

**Comment by yykamei** [comment]
Oh, it's surprising. Thank you :pray:

**Comment by ybiquitous** [comment]
I've opened PR #2195 for PoC.

**Pull request merged**